src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * Source code is available at <https://github.com/FelixBaensch/MORTAR>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package de.unijena.cheminf.mortar.gui.controls;
+
+import javafx.animation.FillTransition;
+import javafx.animation.ParallelTransition;
+import javafx.animation.TranslateTransition;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.control.Control;
+import javafx.scene.effect.DropShadow;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Rectangle;
+import javafx.util.Duration;
+
+/**
+ * A toggle switch to en- and disable features in settings.
+ *
+ * @author Zeynep Dagtekin
+ * @version 1.0.0.0
+ */
+public class ToggleSwitch extends Control {
+    //<editor-fold desc="private final class constants" defaultstate="collapsed">
+    /**
+     * Button.
+     */
+    private final Circle switchButton;
+    /**
+     * Background of the switch.
+     */
+    private final Rectangle switchBackground;
+    /**
+     * Boolean property to keep track of the state.
+     */
+    private final SimpleBooleanProperty switchedOn;
+    /**
+     * transition of the Circle from one side to the other.
+     */
+    private final TranslateTransition switchAnimation;
+    /**
+     * Color transition.
+     */
+    private final FillTransition fillAnimation;
+    /**
+     * Parallel transition of color and Circle.
+     */
+    private final ParallelTransition switchTransition;
+    //</editor-fold>
+    /**
+     * Constructor.
+     */
+    public ToggleSwitch() {
+        /**
+         * code inspired by "JavaFX UI: iOS Style Toggle Switch", uploaded by Almas Baimagambetov on YouTube
+         * https://youtu.be/maX5ymmQixM?si=v2ULa57-pjCmoQlf, 05/17/2024, 10:33
+         */
+        super();
+        this.switchedOn = new SimpleBooleanProperty(false);
```

Comment:
Would be nice to turn the initial state into a constructor parameter and implement a separate parameter-less constructor calling this one here with a defined initial state (off/false).